### PR TITLE
Make Spacy and Gensim dependencies optional

### LINKED ISSUE #307: Make NLP dependencies optional
Some dependencies are only needed in some images, we should make some dependencies optional so that they are only installed when necessary.

For example, Spacy is the biggest dependency, but it is currently not really used by any image.

### PR DESCRIPTION
**Pull request by Lothiraldan**
Fixes #307.

### CODE REVIEW

File: bugbug/nlp.py [hunk 1]
```diff
@@ -3,17 +3,38 @@
...
+try:
+    import spacy
+
+    from spacy.tokenizer import Tokenizer
+    from gensim.models import KeyedVectors
+except ImportError:
+    spacy = None
+
+try:
+    nlp = spacy.load("en_core_web_sm")
```
Comment (marco-c) [inline bugbug/nlp.py]: We should only do this when spacy is available

File: bugbug/nlp.py [hunk 2]
```diff
@@ -3,17 +3,38 @@
...
 from sklearn.feature_extraction.text import TfidfVectorizer
-from spacy.tokenizer import Tokenizer
 
-nlp = spacy.load("en_core_web_sm")
+# Spacy is an optional dependency
+try:
+    import spacy
+
+    from spacy.tokenizer import Tokenizer
+    from gensim.models import KeyedVectors
```
Comment (marco-c) [inline bugbug/nlp.py]: We should also detect when gensim is not available.

File: bugbug/nlp.py (hunk 1 above)
Comment (Lothiraldan) [reply bugbug/nlp.py]: Indeed, it should be fixed now

File: bugbug/nlp.py (hunk 2 above)
Comment (Lothiraldan) [reply bugbug/nlp.py]: Indeed, I introduced a new atomic boolean for this effect

File: bugbug/nlp.py [hunk 3]
```diff
@@ -3,17 +3,40 @@
 # License, v. 2.0. If a copy of the MPL was not distributed with this file,
 # You can obtain one at http://mozilla.org/MPL/2.0/.
 
+from __future__ import print_function
```
Comment (marco-c) [inline bugbug/nlp.py]: This should not be needed, right?

File: bugbug/nlp.py [hunk 4]
```diff
@@ -3,17 +3,40 @@
...
+
+    HAS_OPTIONAL_DEPENDENCIES = True
+except ImportError:
+    pass
+
+try:
+    if HAS_OPTIONAL_DEPENDENCIES:
+        nlp = spacy.load("en_core_web_sm")
+except OSError:
+    msg = "Spacy model is missing, install it with: %s -m spacy download en_core_web_sm"
```
Comment (marco-c) [inline bugbug/nlp.py]: Nit: use f-string here and in OPT_MSG_MISSING

File: bugbug/nlp.py [hunk 5]
```diff
@@ -25,6 +48,11 @@ def spacy_token_lemmatizer(text):
 
 class SpacyVectorizer(TfidfVectorizer):
     def __init__(self, *args, **kwargs):
+
+        # Detect when the optional dependency is missing
```
Comment (marco-c) [inline bugbug/nlp.py]: Nit: `# Detect when the spacy optional dependency is missing` (at some point we might split Spacy and Gensim into separate optional deps)

File: bugbug/nlp.py [hunk 6]
```diff
@@ -37,6 +65,10 @@ def get_word_embeddings():
 
 class MeanEmbeddingTransformer(BaseEstimator, TransformerMixin):
     def __init__(self):
+        # Detect when the optional dependency is missing
```
Comment (marco-c) [inline bugbug/nlp.py]: Nit: `# Detect when the gensim optional dependency is missing` (at some point we might split Spacy and Gensim into separate optional deps)

**Pull request merged**